flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/validation/DefaultValidatorTest.java

```diff
@@ -248,13 +252,40 @@ public void testValidationWithoutDefaultConfig() {
         testError(
                 dep -> dep.getSpec().getJobManager().getResource().setMemory("invalid"),
                 "JobManager resource memory parse error");
-
         testError(
                 dep -> dep.getSpec().getTaskManager().getResource().setMemory(null),
-                "TaskManager resource memory must be defined");
+                "TaskManager memory configuration failed");
```

Comment:
I think the previous error message was much more informative for the user. 
"TaskManager memory configuration failed" is pretty hard to understand.

Reply:
It is thrown by `IllegalConfigurationException` - [JobManagerProcessUtils.java#L7](https://github.com/apache/flink/blob/5c3658aa06b79e8039043145560a1ad2bcce68b0/flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/JobManagerProcessUtils.java#L78)

Full error message is very descriptive, example:

```
TaskManager memory configuration failed: Either required fine-grained memory 
(taskmanager.memory.task.heap.size and taskmanager.memory.managed.size), 
or Total Flink Memory size (Key: 'taskmanager.memory.flink.size' , default: null (fallback keys: [])), 
or Total Process Memory size (Key: 'taskmanager.memory.process.size' , 
default: null (fallback keys: [])) need to be configured explicitly.
```

Reply:
In this case I would expect an error instructing the user to set `spect.taskmanager.resource.memory` instead of the flink configs directly.

Reply:
Changed it to `TaskManager resource memory must be defined using 'spec.taskManager.resource.memory'`.